cli/flox/src/utils/dialog.rs

```diff
@@ -120,6 +129,51 @@ impl<'a, T: Display> Dialog<'a, Select<T>> {
     }
 }
 
+impl<'a, F: FnOnce() -> T + Send, T: Send> Dialog<'a, Spinner<F>> {
+    pub fn spin_with_timeout(self, timeout: Duration) -> T {
```

Comment:
```suggestion
    pub fn spin_if_long_running(self, start_spinning_after: Duration) -> T {
```
Suggestion: Tweak the naming to make it clear that nothing is cancelled after a certain period of time. `timeout` often implies that something has until `timeout` to complete, and after that it will be cancelled. That's not what happens here.